### DISCUSSION TITLE
MissingSchemaError: Schema hasn't been registered for model "Comments". Use mongoose.model(name, schema)

### DISCUSSION THREAD

**Discussion by alshahabrezvi** [vercel/next.js#55857] [Help]
### **why i am geting MissingSchemaError: Schema hasn't been registered for model "Comments".
Use mongoose.model(name, schema) this error .Can anyone help me. I have given my code below:**:::::

import mongoose from "mongoose";

const commentSchema = new mongoose.Schema({
  comment: {
    type: String,
    required: [true, "Please provide a comment"],
  },

  blog: {
    type: mongoose.Types.ObjectId,
    ref: "Blog",
  },
  user: {
    type: mongoose.Types.ObjectId,
    ref: "Users",
  },
});

var Comment =
  mongoose.models.Comments || mongoose.model("Comments", commentSchema);

export default Comment;

import mongoose from "mongoose";

const userSchema = new mongoose.Schema({
    username: {
        type: String,
        required: [true, "Please provide a username"],
        unique: true,
    },
    email: {
        type: String,
        required: [true, "Please provide a email"],
        unique: true,
    },
    password: {
        type: String,
        required: [true, "Please provide a password"],
    },blogs:[{
        type:mongoose.Types.ObjectId,
        ref:'Blog'
        
      }],
    user:{
        type:mongoose.Types.ObjectId,
        ref:'User'
        
      },
   
    comment:{
        type:mongoose.Types.ObjectId,
        ref:'Comments'
        
      },
   
 
})

const User = mongoose.models.Users || mongoose.model("Users", userSchema);

export default User;

import mongoose from "mongoose";

const blogSchema = new mongoose.Schema({
  title: {
    type: String,
    required: [true, "Please provide a title"],
  },
  desc: {
    type: String,
    required: [true, "Please provide a desc"],
  },
  image: {
    type: String,
    required: [true, "Please provide a image"],
  },
  catagory: [
    {
      type: String,
    },
  ],
  users: {
    type: mongoose.Types.ObjectId,
    ref: "Users",
  },

  comments: [{ type: mongoose.Types.ObjectId, ref: "Comments" }],
  date: { type: Date, default: Date.now },
});

var Blogs = mongoose.models.Blog || mongoose.model("Blog", blogSchema);
export default Blogs;

route.js file

import { NextResponse, NextRequest } from "next/server";
import { connect } from "@/dbConfig/dbConfig";
import Blogs from "@/models/blogModel";

connect();

export async function POST(request, { params }) {
  try {
    const { id } = params;

    const blog = await Blogs.find({ _id: id }).populate({
     path: 'comments',
     model:'Comments',
     populate:{
      path:'user',
      model:'Users'
     }
    }

    );
   

    if (!blog) {
      return NextResponse.json({ message: "No blog" });
    }
    return NextResponse.json({
      message: "all blog",
      blog,
      success: true,
      status: 200,
    });
  } catch (error) {
    console.log(error);
    return NextResponse.json({
      message: "there is an problem",
      success: false,
      status: 404,
      error,
    });
  }
}

### Additional information

```js
i am new in web development please experts help me tnq ;(
```

**Comment by bikramdhami0100** [comment]
**why i am geting MissingSchemaError: Schema hasn't been registered for model "Comments". 
in route.js or route.ts  import Comments and the inside try block  const commentregister=await Comments.find() ; or simply write const commentregister=await Comments . it's definitelly work
